s3cache.go

```diff
@@ -19,22 +19,25 @@ import (
 	"golang.org/x/crypto/acme/autocert"
 )
 
-// Logger logs.
+// Logger for outputing logs
 type Logger interface {
 	Printf(format string, v ...interface{})
 }
 
+// Making sure that we're adhering to the autocert.Cache interface
 var _ autocert.Cache = (*Cache)(nil)
 
-// Cache provides an autocert s3 cache.
+// Cache provides a s3 backend to the autocert cache.
 type Cache struct {
 	bucket string
 	s3     s3iface.S3API
 
+	Prefix string // remember to add the trailing slash
+
 	Logger Logger
 }
 
-// New creates a autocert s3 cache.
+// New creates an s3 interface to the autocert cache.
```

Comment:
I wouldn't really call it an interface, the doc changes you made at `Cache` seem to fit better.